ext/ScreenShot/ScreenShot.h

```diff
@@ -5,27 +5,151 @@
-{
-	// will change this, https://github.com/buildaworldnet/IrrlichtBAW/issues/148
-	if (isBlockCompressionFormat(source->getCreationParameters().format))
-		return nullptr;
+	namespace ext
+	{
+		namespace ScreenShot
+		{
+			/*
+				Creates useful FBO that may be used for instance 
+				to fetch default render target color data attachment 
+				after rendering scene.
 
-	auto extent = source->getMipSize(sourceMipLevel);
-	video::IGPUImage::SBufferCopy pRegions[1u] = { {destOffset,extent.x,extent.y,{static_cast<asset::IImage::E_ASPECT_FLAGS>(0u),sourceMipLevel,0u,1u},{0u,0u,0u},{extent.x,extent.y,extent.z}} };
-	driver->copyImageToBuffer(source,destination,1u,pRegions);
+				The usage is following:
 
-	return driver->placeFence(implicitflush);
-}
+				- create frame buffer using the function
+				- blit framebuffers, driver->blitRenderTargets(nullptr, frameBuffer, false, false);
+				Note that in the call above we don't want to copy depth and stencil buffer, but event though default
+				FBO contains depth buffer.
+				- pass frame buffer to performScreenShot(video::IFrameBuffer*)
+
+				Notes:
+
+				- color buffer is placed under video::EFAP_COLOR_ATTACHMENT0 attachment
+				- depth buffer is placed under video::EFAP_DEPTH_ATTACHMENT attachment
+			*/
+
+			irr::video::IFrameBuffer* createDefaultFBOForScreenshoting(core::smart_refctd_ptr<IrrlichtDevice> device)
+			{
+				auto driver = device->getVideoDriver();
+
+				auto createAttachement = [&](bool colorBuffer)
+				{
+					asset::ICPUImage::SCreationParams imgInfo;
+					imgInfo.format = colorBuffer ? asset::EF_R8G8B8A8_SRGB : asset::EF_D24_UNORM_S8_UINT;
+					imgInfo.type = asset::ICPUImage::ET_2D;
+					imgInfo.extent.width = driver->getCurrentRenderTargetSize().Width;
+					imgInfo.extent.height = driver->getCurrentRenderTargetSize().Height;
+					imgInfo.extent.depth = 1u;
+					imgInfo.mipLevels = 1u;
+					imgInfo.arrayLayers = 1u;
+					imgInfo.samples = asset::ICPUImage::ESCF_1_BIT;
+					imgInfo.flags = static_cast<asset::IImage::E_CREATE_FLAGS>(0u);
+
+					auto image = asset::ICPUImage::create(std::move(imgInfo));
+					const auto texelFormatBytesize = getTexelOrBlockBytesize(image->getCreationParameters().format);
+
+					auto texelBuffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(image->getImageDataSizeInBytes());
+					auto regions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<asset::ICPUImage::SBufferCopy>>(1u);
+					asset::ICPUImage::SBufferCopy& region = regions->front();
+
+					region.imageSubresource.mipLevel = 0u;
+					region.imageSubresource.baseArrayLayer = 0u;
+					region.imageSubresource.layerCount = 1u;
+					region.bufferOffset = 0u;
+					region.bufferRowLength = image->getCreationParameters().extent.width;
+					region.bufferImageHeight = 0u;
+					region.imageOffset = { 0u, 0u, 0u };
+					region.imageExtent = image->getCreationParameters().extent;
+
+					image->setBufferAndRegions(std::move(texelBuffer), regions);
```

Comment:
you dont need to set buffer and regions (no data to upload anyway).

Here's a prime example of an API where you don't need to use `asset::` at all, just create the IGPUImage and IGPUImageView directly without conversions and CPU objects